lib/charms/zookeeper/v0/client.py

```diff
@@ -425,6 +425,24 @@ def delete_znode_leader(self, path: str) -> None:
         ) as zk:
             zk.delete_znode(path=path)
 
+    def get_version(self) -> str:
+        """Get ZooKeeper version.
+
+        Version will have a similar format to
+            3.6.4--d65253dcf68e9097c6e95a126463fd5fdeb4521c, built on 12/18/2022 18:10 GMT
+        """
+        with ZooKeeperClient(
+            host=self.leader,
+            client_port=self.client_port,
+            username=self.username,
+            password=self.password,
+            use_ssl=self.use_ssl,
+            keyfile_path=self.keyfile_path,
+            keyfile_password=self.keyfile_password,
+            certfile_path=self.certfile_path,
+        ) as zk:
+            return zk.srvr["Zookeeper version"]
```

Comment:
**todo:** We should just get the version as I don't think we'll care about the rest.

```python
full_version = zk.srvr["ZooKeeper version"]

if not (match := re.match(string=full_version, pattern=r"([0-9\.]+)")):
    return None

return match[0]
```

Reply:
Sounds good, I was doing this on Kafka side